planner/plansdk/plansdk.go

```diff
@@ -116,6 +108,10 @@ func (p *Plan) WithError(err error) *Plan {
 	return p
 }
 
+func (p *Plan) ShellAndDevPackages() []string {
+	return pkgslice.Unique(append(p.DevPackages, p.ShellPackages...))
```

Comment:
You probably want to copy to a new slice here, otherwise the returned slice and DevPackages will share the same backing array.